scripts/gz3d/build_parent_sample.py

```diff
@@ -0,0 +1,172 @@
+    gz_spiral_fraction = table["gz_spiral_votes"] / table["gz_total_classifications"]
+    mask = (gz_bar_fraction>=threshold) | (gz_spiral_fraction>=threshold)
+    return mask 
+
+def is_link_to_fits_file(link_text, endswith='.fits.gz'):
+    if link_text is None:
+        return False
+    else:
+        return link_text.endswith(endswith)
+
+def build_filelist(page_url: str, limit: int = None):
+    """ Build a list of relevant files to transfer """
+    response = requests.get(page_url)
+    soup = BeautifulSoup(response.content, 'html.parser')
+    hrefs = soup.find_all('a', string=is_link_to_fits_file)
+    # Get href attribute from each element as list
+    files = []
+
+    n_files = limit or len(hrefs)
+    for ref in hrefs:
+        files.append(ref['href'])
+        if len(files) >= n_files:
+            break
+    
+    return files
+
+def download_all_files(links, base_url, test=False, dir="./data", force=False):
+    for link in tqdm(links, desc="Downloading files"):
+        try:
+            download_file(base_url, link, download_dir=dir, force=force)
+        except Exception as e:
+            print(f"Error downloading {urljoin(base_url, link)}: {e}")
+        if test:
+            break
+
+def download_file(url: str, file_name: str, download_dir: str ='./data', chunk_size: int = 8192, force: bool = False):
+    """Download a file from a URL to a local directory in chunks (streaming).
+
+    Args:
+        url (str): URL of the file to download.
+        file_name (str): Name of the file to download.
+        download_dir (str, optional): Path to which the data is downloaded. Defaults to './data'.
+        chunk_size (int, optional): Chunk size for streaming. Defaults to 8192.
+
+    Returns:
+        str: Path to the downloaded file.
+    """
+
+    full_url = urljoin(url, file_name)
+
+    local_filename = download_dir / pathlib.Path(file_name)
+    if local_filename.exists() and not force:
+        return str(local_filename)
+    local_filename.parent.mkdir(parents=True, exist_ok=True)
+
+    with requests.get(full_url, stream=True) as r:
+        r.raise_for_status()
+        with local_filename.open('wb') as f:
+            for chunk in r.iter_content(chunk_size=chunk_size):
+                if chunk: # Skip empty chunks
+                    f.write(chunk)
+    
+    return str(local_filename)  # Return the file path as a string
+
+
+def _read_fits_file(loc):
+    with fits.open(loc) as hdul:
+        entry = pd.DataFrame(fits.open(loc)[5].data) 
+        entry.columns = map(str.lower, entry.columns) # lowercase column names
+        entry = Table.from_pandas(entry)
+
+        entry['false_color'] = hdul[0].data.astype(np.uint8)[None, ...]
+        entry['center'] = hdul[1].data.astype(np.uint8)[None, ...]
+        entry['star'] = hdul[2].data.astype(np.uint8)[None, ...]
+        entry['spiral'] = hdul[3].data.astype(np.uint8)[None, ...]
+        entry['bar'] = hdul[4].data.astype(np.uint8)[None, ...]
+    return entry
+
```

Comment:
Add helper function to flip all fields vertically:
```suggestion


def _vertical_flip_fields(entry):
    """Vertically flip fields to be consistent with the orientation of `legacysurvey` fields."""
    entry["false_color"] = np.flipud(entry["false_color"])
    entry["center"] = np.flipud(entry["center"])
    entry["star"] = np.flipud(entry["star"])
    entry["spiral"] = np.flipud(entry["spiral"])
    entry["bar"] = np.flipud(entry["bar"])
    return entry


```